design/src/rfcs/rfc0020_service_builder.md

```diff
@@ -283,4 +280,269 @@ let new_routes = old_routes
+
+#### Routing
+
+The Smithy model not only specifies the `http::Request` decomposition and `http::Response` composition for a given service, it also determines the routing. The `From<OperationRegistry>` implementation, described in [Builder](#builder), yields a fully formed router based on the protocol and [http traits](https://awslabs.github.io/smithy/1.0/spec/core/http-traits.html#http-trait) specified.
+
+This is in contrast to `axum`, where the user specifies the routing by use of various combinators included on the `axum::Router`, applied to other `tower::Service`s. In an `axum` application one might encounter the following code:
+
+```rust
+let user_routes = Router::new().route("/:id", /* service */);
+
+let team_routes = Router::new().route("/", /* service */);
+
+let api_routes = Router::new()
+    .nest("/users", user_routes)
+    .nest("/teams", team_routes);
+
+let app = Router::new().nest("/api", api_routes);
+```
+
+Note that, in `axum` handlers are quickly converted to `tower::Service` (via `IntoService`) before they are passed into the `Router`. In contrast, in `smithy-rs` handlers are passed into a builder and then the conversion to `tower::Service` is performed (via `OperationHandler`).
+
+Introducing state to handlers in `axum` is done in the same way as `smithy-rs`, described briefly in [Handlers](#handlers) - a layer is used to insert state into incoming `http::Request`s and the `Handler` implementation pops it out of the type map layer. In `axum`, if a customer wanted to scope state to all routes within `/users/` they are able to do the following:
+
+```rust
+async fn handler(Extension(state): Extension</* State */>) -> /* Return Type */ {}
+
+let api_routes = Router::new()
+    .nest("/users", user_routes.layer(Extension(/* state */)))
+    .nest("/teams", team_routes);
+```
+
+In `smithy-rs` a customer is only able to apply a layer to either the `aws_smithy_http::routing::Router` or every route via the [layer method](#router) described above.
+
+# Proposal
+
+The proposal is presented as a series of compatible transforms to the existing service builder, each paired with a motivation. Most of these can be independently implemented, but in the case where there exists an interdependency it is stated.
+
+Although presented as a mutation to the existing service builder, the actual implementation should exist as a entirely separate builder - reusing code generation from the old builder but exposes a new Rust API. Preserving the old API surface will prevent breakage and make it easier to perform comparative benchmarks and testing.
+
+## Remove two-step build procedure
```

Comment:
I assume this transformation will be superseded by the yet-to-be-drafted "Type erasure with the name of the Smithy service" (?) They are both independent, but since they're also both breaking changes, I'd just release one of them. If this is the case I'd note it down somewhere in the RFC.